File: banda/SergioDeIscar-banda-java/SergioDeIscar-banda-java/src/main/java/org/example/models/Cantante.java
package org.example.models;

import org.example.enums.TonoType;
import org.example.interfaces.ICantante;

public final class Cantante extends Musician implements ICantante {
    private TonoType tonoType;
    private final Float salario = super.salario * 1.4f;

    public Cantante(String nombre, int experiencia, TonoType tono) {
        super(nombre, experiencia);
        setTono(tono);
    }

    @Override
    public float getSalario() {
        return salario;
    }

    @Override
    public TonoType getTono() {
        return tonoType;
    }

    @Override
    public void setTono(TonoType tono) {
        tonoType = tono;
    }

    @Override
    public void cantar() {
        System.out.println("El cantante " + nombre + " esta cantando");
    }

    @Override
    public String getNombre() {
        return this.nombre;
    }
}

File: banda/SergioDeIscar-banda-kotlin/SergioDeIscar-banda-kotlin/src/main/kotlin/models/Cantante.kt
package bandamusical.models

import bandamusical.interfeces.ICantante

class Cantante(nombre:String,
               override val experiencia: Int,
               override val tono: TonoType): Musician(nombre), ICantante {
    override val salario: Float = super.salario * 1.4f
    override fun respirar() {
        "El músico cantante $nombre respira."
    }

    override fun cantar() {
        println("El cantante $nombre canta con un tono $tono.")
    }
}

enum class TonoType{
    CONTRALTO,
    MEZZOSOPRANO,
    SOPRANO,
    BAJO,
    BARITONO, //BARÍTONO
    TENOR,
    CONTRATENOR
}